runtime/queries/css/highlights.scm

```diff
@@ -1,64 +1,86 @@
 (comment) @comment
 
-(tag_name) @tag
-(nesting_selector) @tag
-(universal_selector) @tag
+[
+ (tag_name)
+ (nesting_selector)
+ (universal_selector)
+] @tag
 
-"~" @operator
-">" @operator
-"+" @operator
-"-" @operator
-"*" @operator
-"/" @operator
-"=" @operator
-"^=" @operator
-"|=" @operator
-"~=" @operator
-"$=" @operator
-"*=" @operator
+[
+ "~"
+ ">"
+ "+"
+ "-"
+ "*"
+ "/"
+ "="
+ "^="
+ "|="
+ "~="
+ "$="
+ "*="
+] @operator
 
-"and" @operator
-"or" @operator
-"not" @operator
-"only" @operator
-
-(attribute_selector (plain_value) @string)
-(pseudo_element_selector (tag_name) @attribute)
-(pseudo_class_selector (class_name) @attribute)
-
-(class_name) @variable.other.member
-(id_name) @variable.other.member
-(namespace_name) @variable.other.member
-(property_name) @variable.other.member
-(feature_name) @variable.other.member
-
-(attribute_name) @attribute
-
-(function_name) @function
+[
+ "and"
+ "not"
+ "only"
+ "or"
+] @keyword.operator
 
 ((property_name) @variable
  (#match? @variable "^--"))
 ((plain_value) @variable
  (#match? @variable "^--"))
 
-"@media" @keyword
-"@import" @keyword
-"@charset" @keyword
-"@namespace" @keyword
-"@supports" @keyword
-"@keyframes" @keyword
-(at_keyword) @keyword
-(to) @keyword
-(from) @keyword
-(important) @keyword
+(attribute_name) @attribute
+(class_name) @label
+(feature_name) @variable.other.member
+(function_name) @function
+(id_name) @label
+(namespace_name) @namespace
+(property_name) @variable.other.member
+
+[
+ "@charset"
+ "@import"
+ "@keyframes"
+ "@media"
+ "@namespace"
+ "@supports"
+ (at_keyword)
+ (from)
+ (important)
+ (to)
+] @keyword
+
+[
+ "#"
+ "."
+] @punctuation
 
 (string_value) @string
+((color_value) "#") @string.special
 (color_value) @string.special
 
 (integer_value) @constant.numeric.integer
 (float_value) @constant.numeric.float
-(unit) @type
+(unit) @constant.numeric
```

Comment:
Why the change here? `unit` is for types like  "rem" in `1.6rem`

Reply:
The idea was that with the highlighting of many other languages any number type suffixes and prefixes adopt the same colour as the number they're joined to, e.g. `123u`, `0.5f`, `43m` (C#), `23.4f32`, `16u8` (Rust) etc. It's also how CSS numbers are highlighted in some other editors, like VSCode. Different colours without whitespace between them looks a bit jarring to me too, but that's subjective

Reply:
Hmm yeah that does make sense. I don't feel very strongly about this. It seems like some highlighters do diverge from the normal and use a different color for units like tree-sitter-css repo's highlights (https://github.com/tree-sitter/tree-sitter-css/blob/769203d0f9abe1a9a691ac2b9fe4bb4397a73c51/queries/highlights.scm#L60) or nvim-treesitter (https://github.com/nvim-treesitter/nvim-treesitter/blob/1656f67f261ab51525288d0c4b38e1577f79eaf3/queries/css/highlights.scm#L66) or github (which is probably using tree-sitter-css) but most use the same color as the value (kakoune, vim, emacs, w3c docs, mdn docs).

Let's go with `constant.numeric` 👍

Reply:
Just for clarification, are you suggesting we should allow the unit to follow the colour of the number it's joined to, in which case the `(unit)` tag should be omitted entirely, or that it should be set to `constant.numeric` so that it could be a different colour if people want it to be?

Reply:
Oh, I missed that omitting the `unit` stanza highlights it following the number. This looks good then 👍